src/components/layerlist/LayerLegendImage.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+    <!-- Remarks: 
+    As we need none of the responsive functionality of v-img, use a simple
+    HTLM img to stop the component from flickering, when the image is re-requested. 
+    -->
+    <img v-if="legendURL" :src="legendURL">
+</template>
+
+<script>
+
+import LayerLegend from '../../util/LayerLegend';
+
+/**
+ * Module for one legend element.
+ */
+export default {
+  name: 'wgu-layerlegendimage',
+  props: {
+    mapView: { type: Object, required: true },
+    layer: { type: Object, required: true }
+  },
+  data () {
+    return {
+      resolution: this.mapView.getResolution(),
+      viewResolutionChanged: undefined
+    }
+  },
+
+  /**
+   * Register for an event to update the legend on resolution change.
+   */
+  created () {
+    const viewResolutionChanged = function (event) {
```

Comment:
why not creating a method in the `methods` block?

Reply:
The problem that you often run into is binding the this scope. You have to be careful then to also mess up event unregistration, if I remember correcly. So I didn`t bother wrapping it up for a one-liner.
The pattern has been used elsewhere too, e.g.  in the [Progress Component](https://github.com/wegue-oss/wegue/blob/76fa53070e9ce615c9fcdf2c10bc280c576228eb/src/components/progress/MapLoadingStatus.vue#L62)

Want me to try and change it?

Reply:
thanks for the explanation. We can keep it like it is.